bake/bake.go

```diff
@@ -232,7 +232,7 @@ func ReadTargets(ctx context.Context, files []File, targets, overrides []string,
 	}
 
 	for _, target := range targets {
-		if target == "default" {
+		if _, ok := n["default"]; ok && target == "default" {
```

Comment:
would be beneficial to rename `n` to `groupsMap` or `mGroups` as it starts to be quite hard to read now.

Single-letter vars are ok if the variable is locally scoped for a small block after its definition(temp variable).